tests/test_transition_matrix_solver.py

```diff
@@ -0,0 +1,230 @@
+import numpy as np
+import pytest
+
+from elexsolver.TransitionMatrixSolver import TransitionMatrixSolver
+
+RTOL = 1e-04
+ATOL = 1e-04
+
+
+def test_matrix_fit_predict():
+    X = np.array(
+        [
+            [1, 2],
+            [3, 4],
+            [5, 6],
+            [7, 8],
+            [9, 10],
+            [11, 12],
+        ]
+    )
+
+    Y = np.array(
+        [
+            [2, 3],
+            [4, 5],
+            [6, 7],
+            [8, 9],
+            [10, 11],
+            [12, 13],
+        ]
+    )
+
+    expected = np.array([[0.760428, 0.239572], [0.216642, 0.783358]])
+
+    tms = TransitionMatrixSolver()
+    current = tms.fit_predict(X, Y)
+    np.testing.assert_allclose(expected, current, rtol=RTOL, atol=ATOL)
+
+
+def test_matrix_fit_predict_with_weights():
+    X = np.array(
+        [
+            [1, 2],
+            [3, 4],
+            [5, 6],
+            [7, 8],
+            [9, 10],
+            [11, 12],
+        ]
+    )
+
+    Y = np.array(
+        [
+            [2, 3],
+            [4, 5],
+            [6, 7],
+            [8, 9],
+            [10, 11],
+            [12, 13],
+        ]
+    )
+
+    weights = np.array([500, 250, 125, 62.5, 31.25, 15.625])
+
+    expected = np.array([[0.737329, 0.262671], [0.230589, 0.769411]])
```

Comment:
I'm curious where this expected is from?

Reply:
It's the result of running the code on the next few lines below this.  Do you get a different result if you run this locally? 😬

Reply:
no, I was just curious. This isn't super ideal (for quantile regression the expected solution was taken from an implementation in R) but I guess we have no other option here.